### PR TITLE
FIXES: #402 =>Add transition and overlay for Dialog/Modal

### PR DESCRIPTION
**Pull request by SMahil**
For Dialog, Transition has been added and overlay has been set to the body. Earlier it was just showing or hiding the dialog box.

fixes  https://github.com/almighty/almighty-ui/issues/402

### CODE REVIEW

**Comment by joshuawilson** [comment]
@SMahil this has failing tests. Please rebase and then maybe @naina-verma can help you.

**Comment by SMahil** [comment]
@joshuawilson rebase done and tests are resolved

File: src/app/shared-component/dialog/dialog.component.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {
   EventEmitter,
   Input,
   OnInit,
-  Output
+  Output, trigger, style , animate, state, transition
```
Comment (sanbornsen) [inline src/app/shared-component/dialog/dialog.component.ts]: Please write them in order and vertically.

File: src/app/shared-component/dialog/dialog.component.ts [hunk 2]
```diff
@@ -12,6 +12,28 @@ import { Dialog } from './dialog';
...
+        transform: 'translateY(0)'
+      })),
+      transition('inactive <=>active', animate('500ms ease all'))
+    ]),
+    trigger('modalOverlay', [
+      state('inactive', style({
+        background: 'transparent'
+      })),
+      state('active', style({
+        background: 'rgba(0,0,0,0.4)'
```
Comment (SMahil) [inline src/app/shared-component/dialog/dialog.component.ts]: both have same output. If we are not following rgba we can change to hex with opacity

Comment (sanbornsen) [inline src/app/shared-component/dialog/dialog.component.ts]: @mindreeper2420 it's a background over lay that comes under the modal.

Comment (mindreeper2420) [inline src/app/shared-component/dialog/dialog.component.ts]: @SMahil just a consistency bit, IMO, but not a problem :)

@sanbornsen you can set those with opacity effects and layering, so makes no difference

Comment (SMahil) [inline src/app/shared-component/dialog/dialog.component.ts]: hex+alpha is not working over here

**Comment by joshuawilson** [comment]
@SMahil please rebase

File: src/app/footer/footer.component.html [hunk 3]
```diff
@@ -1 +1 @@
-<div id="footer_container" class="footer width100 pull-left dn">Copyright 2016 Red Hat</div>
+<div id="footer_container" class="footer col-md-12 col-sm-12 col-xs-12 pull-left dn">Copyright 2016 Red Hat</div>
```
Comment (nimishamukherjee) [inline src/app/footer/footer.component.html]: Do we need the footer, we always have quick add at the bottom.

Comment (SMahil) [inline src/app/footer/footer.component.html]: I just kept it rather than removing if we need it later sometime

**Comment by joshuawilson** [comment]
@SMahil what issue/task is the suppose to fix? Please add the issue number to the Description at the top. Also please explain what this fix does?

**Pull request merged**